File: src/ua/nure/lab2/PotholerBehaviour.java
package ua.nure.lab2;

import jade.core.behaviours.FSMBehaviour;

public abstract class PotholerBehaviour extends FSMBehaviour {

    public static final String STATE_REGISTER = "Register";
    public static final String STATE_PLAY = "Play";

    public static final int REGISTERED = 0;
    public static final int OVER = 1;
    public static final int NOT_REGISTERED = 2;

    abstract void before();

    abstract void after();

}

File: MSiT_Lab3/src/main/scala/wumpus/PotholerBehaviour.scala
package wumpus

import jade.core.behaviours.FSMBehaviour

abstract class PotholerBehaviour extends FSMBehaviour {
  val STATE_REGISTER = "Register"
  val STATE_PLAY = "Play"

  val REGISTERED = 0
  val OVER = 1
  val NOT_REGISTERED = 2

  private[wumpus] def before(): Unit

  private[wumpus] def after(): Unit
}
